wiki/markdown/generate-service.md

```diff
@@ -0,0 +1,617 @@
+```
+
+### Examples for `/event_types/{mp}` endpoint
+
+#### Retrieve the event types:
+
+```
+curl -X GET --header 'Accept: application/json' 'http://localhost:8080/generate-service/event_types/eiffelsemantics'
+```
+
+Result:
+
+```
+["EiffelArtifactPublishedEvent","EiffelActivityFinishedEvent","EiffelActivityCanceledEvent","EiffelArtifactCreatedEvent","EiffelActivityTriggeredEvent","EiffelConfidenceLevelModifiedEvent","EiffelActivityStartedEvent","EiffelAnnouncementPublishedEvent","EiffelCompositionDefinedEvent","EiffelTestCaseCanceledEvent","EiffelTestCaseTriggeredEvent","EiffelTestExecutionRecipeCollectionCreatedEvent","EiffelEnvironmentDefinedEvent","EiffelFlowContextDefinedEvent","EiffelSourceChangeCreatedEvent","EiffelSourceChangeSubmittedEvent","EiffelTestCaseFinishedEvent","EiffelTestCaseStartedEvent","EiffelTestSuiteFinishedEvent","EiffelTestSuiteStartedEvent","EiffelIssueVerifiedEvent","EiffelArtifactReusedEvent","EiffelServiceStoppedEvent","EiffelServiceStartedEvent","EiffelServiceReturnedEvent","EiffelServiceDiscontinuedEvent","EiffelServiceDeployedEvent","EiffelServiceAllocatedEvent","EiffelArtifactDeployedEvent","EiffelAlertAcknowledgedEvent","EiffelAlertCeasedEvent","EiffelAlertRaisedEvent"]
+```
+
+### Examples for `/template/{type}/{mp}` endpoint
+
+#### Retrieve the event template:
+
+```
+curl -X GET --header 'Accept: application/json' 'http://localhost:8080/generate-service/template/EiffelArtifactPublishedEvent/eiffelsemantics'
+```
+
+Result:
+
+```
+{"msgParams":{"meta":{"type":"EiffelArtifactPublishedEvent","version":"1.1.0","tags":[],"source":{"domainId":"","host":"","name":"","uri":""},"security":{"sdm":{"authorIdentity":"required if sdm present","encryptedDigest":"required if sdm present"}}}},"eventParams":{"data":{"locations":[{"type":"required ,can be anyone of ARTIFACTORY,NEXUS,PLAIN,OTHER","uri":"required"}],"customData":[{"key":"required if customData present","value":"required if customData present"}]},"links":[{"type":"required ARTIFACT and CAUSE,CONTEXT,FLOW_CONTEXT are optional","target":"required"}]}}
+```
+
+### Examples for `/versions` endpoint
+
+```
+curl -H "Content-Type: application/json" -X GET http://localhost:8080/generate/versions
+```
+
+Result:
+
+```
+{"serviceVersion":{"version":"x.x.x"},"endpointVersions":{"semanticsVersion":"x.x.x"}}
+```
+
+## Status Codes
+
+The response generated will have internal status codes for each and every event based on the input JSON provided.
+
+Status codes are generated according to the below table.
+
+<table class="wikitable">
```

Comment:
Any reason to not use markdown syntax for these tables? I think it would be easier to maintain. https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet